core/modules/mlfqueue.cc

```diff
@@ -0,0 +1,346 @@
+#include "mlfqueue.h"
+#include "../mem_alloc.h"
+
+#include <rte_ether.h>
+#include <rte_ip.h>
+#include <rte_udp.h>
+#include <time.h>
+
+#include <glog/logging.h>
+
+#include <string>
+#include <cmath>
+#include <iostream>
+#include <fstream>
+
+bool operator==(const MLFQueue::Q_Id &id1, const MLFQueue::Q_Id &id2) {
+  bool ips = id1.src_ip == id2.src_ip && id1.dst_ip == id2.dst_ip;
+  bool ports = id1.src_port == id2.src_port && id1.dst_port == id2.dst_port;
+  return ips && ports && id1.protocol == id2.protocol;
+}
+bool operator<(const MLFQueue::Q_Id &id1, const MLFQueue::Q_Id &id2)  {
+  bool ips = id1.src_ip < id2.src_ip && id1.dst_ip < id2.dst_ip;
+  bool ports = id1.src_port < id2.src_port && id1.dst_port < id2.dst_port;
+  return ips && ports && id1.protocol < id2.protocol;
+}
+
+int RoundToPowerTwo(int val) {
+ return pow(2, ceil(log2(val)));
```

Comment:
This will probably be slow, so you don't want to be executing it on the fast path.